backend/src/jobs_server/models.py

```diff
@@ -103,3 +104,20 @@ def from_managed_workload(cls, workload: "ManagedWorkload") -> Self:
             spec=workload.spec,
             kueue_status=workload.status,
         )
+
+
+class LogsParams(BaseModel):
+    stream: bool = Field(default=False, description="Whether to stream the logs")
+    tail: int = Field(
+        default=100, description="Number of tail lines of logs, -1 for all"
+    )
+
+    @field_validator("tail")
+    @classmethod
+    def validate_tail(cls, v):
+        if v < -1:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="tail must be -1 or non-negative integer",
+            )
+        return v
```

Comment:
This can probably be replaced with a [`conint`](https://docs.pydantic.dev/latest/api/types/#pydantic.types.conint)

Reply:
If not, the docs say field validators should return very specific things:

> validators should either return the parsed value or raise a ValueError or AssertionError (assert statements may be used).

Reply:
Unsure here, the docs have a warning that `coint` is discouraged and will be depricated in pydantic 3.0.

I raise the exception here such that the error is propagated to the client. Otherwise, only `500 Server Error` is returned (except, of course, we remove the validation).

Reply:
Update: What do you think about handling the `ValidationError` in a FastAPI `exception_handler` ?  as proposed in c099b?

Reply:
Sounds good! I introduced a helper function to turn any Pydantic model into a dependency in 161a7e6 (based on the discussion [here](https://github.com/fastapi/fastapi/discussions/9071)), which I believe allows multiple validation errors in a single request to be handled correctly (which the exception handler cannot do).